runtime/lua/vim/lsp/util.lua

```diff
@@ -1497,18 +1515,29 @@ do --[[ References ]]
   ---@param bufnr buffer id
   ---@param references List of `DocumentHighlight` objects to highlight
   ---@see https://microsoft.github.io/language-server-protocol/specifications/specification-3-17/#documentHighlight
-  function M.buf_highlight_references(bufnr, references)
+  function M.buf_highlight_references(bufnr, references, client_id)
     validate { bufnr = {bufnr, 'n', true} }
+    local client = vim.lsp.get_client_by_id(client_id)
+
```

Comment:
```suggestion
    local client = vim.lsp.get_client_by_id(client_id)
    if not client then
      return
    end
```


There could be a race between client exiting / receiving the highlights.
I think in this scenario it is fine to just drop the highlights.